processor/spanmetricsprocessor/cache.go

```diff
@@ -0,0 +1,55 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanmetricsprocessor
+
+import lru "github.com/hashicorp/golang-lru"
+
+// Cache is consist of an LRU cache and the evicted items from the LRU cache
+// this data structure makes sure all the cached item can be retrieved either from the LRU cache or the evictedItems map
+type Cache struct {
+	*lru.Cache
```

Comment:
Is it thread safe?

Reply:
It is concurrent safe. The `lru.Cache` has an RWMutex inside.